File: 0905-Sort-Array-By-Parity.js
/**
 * @param {number[]} A
 * @return {number[]}
 */
const sortArrayByParity = function(A) {
  let i = 0;
  let length = A.length;
  let odd;
  
  while (length) {
    if(A[i] % 2) {
      odd = A[i];
      A.splice(i, 1);
      A.push(odd);
      length--;
    } else {
      i++;
      length--;
    }
  }
  
  return A;
};

File: solutions/0905-Sort-Array-By-Parity.cpp
class Solution {
public:
  vector<int> sortArrayByParity(vector<int>& A) {
    vector<int> even(A.size()), odd(A.size()), hey = {1,2,3,4};
    
    auto it1 = copy_if(A.begin(), A.end(), even.begin(),  [](int x) { return (x % 2) == 0; });
    even.resize(distance(even.begin(),it1));
      
    auto it2 = copy_if(A.begin(), A.end(), odd.begin(), [](int x) { return (x % 2) != 0; });
    odd.resize(distance(odd.begin(),it2));
      
    even.insert(even.end(), odd.begin(), odd.end());
    return even;
  }
};
